packages/common/src/ui/components/modals/ConfirmationModal/ConfirmationModalProvider.tsx

```diff
@@ -82,13 +97,18 @@ export const useConfirmationModal = ({
   onConfirm,
   message,
   title,
+  onCancel,
 }: PartialBy<ConfirmationModalState, 'open'>) => {
-  const { setOpen, setMessage, setOnConfirm, setTitle } = useContext(Context);
+  const { setOpen, setMessage, setOnConfirm, setOnCancel, setTitle } =
+    useContext(Context);
 
-  const trigger = () => {
-    setMessage(message);
-    setOnConfirm(onConfirm);
-    setTitle(title);
+  const trigger = (
+    paramPatch?: Partial<PartialBy<ConfirmationModalState, 'open'>>
+  ) => {
```

Comment:
It might be better to force the onChange to be required here 🤔

Reply:
but that makes it hard to use when not passing in a trigger-time version of `onConfirm`. tricky either way..
I prefer what you've done